### PR TITLE
lazily imports more modules for performance

### PR DESCRIPTION
**Pull request by dmalan**
Improves performance of using just `get_*` by lazily loading more of `SQL` and `Flask`.  E.g., on a recent MBP, reduced performance of

```
import cs50

i = cs50.get_int()
print(i)
```

from ~500ms to ~300ms.

Cf. https://wiki.python.org/moin/PythonSpeed/PerformanceTips#Import_Statement_Overhead

We could alternatively break a deliberate abstraction barrier and hardcode awareness of / support for Flask inside of `SQL`, which might further delay expensive imports.

### CODE REVIEW

File: src/cs50/__init__.py [hunk 1]
```diff
@@ -1,6 +1,28 @@
 import os
 import sys
 
+
+class CustomImporter(object):
```
Comment (Jelleas) [inline src/cs50/__init__.py]: Unless I'm missing something, all this does is alias `cs50.sql` to `cs50.SQL`. Flask indirectly import `cs50.sql` anyway, so does this not help in lazy loading.

File: src/cs50/__init__.py [hunk 2]
```diff
@@ -19,8 +41,8 @@
     # Replace Flask's logger
     from . import flask
 
-    # Wrap SQLAlchemy
-    from .sql import SQL
+    # Lazily load CS50.SQL
+    sys.meta_path.append(CustomImporter())
```
Comment (Jelleas) [inline src/cs50/__init__.py]: At this point `cs50.flask` has already imported `cs50.sql`, so there's not much to gain

**Comment by Jelleas** [comment]
There is indeed a significant overhead. From testing it's `Flask` adding most of the work, then `SQLAlchemy` and to a very small degree `sqlparse`. Lazily importing just those shrinks the import time on`cs50` from 300ms to 50ms for me. Lazily importing anything else did not make a difference.

It is `cs50.flask` that adds the most overhead. Sadly I see no way around it, because overwriting the logger is a task that you cannot do lazily. But, is this a problem that needs a hack? I'd say the Pythonic way is to import a submodule separately if it does not need to be loaded. Such as `matplotlib` and `matplotlib.pyplot`, or even `check50` and `check50.c` ;). You would just import `cs50` as normal for `get_*` and such, and import `cs50.flask` once you are working with flask. Conceptually that makes sense to me, because they are actual separate things.

**Comment by kzidane** [comment]
Also as an aside, I think both `find_module` and `load_module` are deprecated fwiw.

**Comment by Jelleas** [comment]
@kzidane you're right! (https://docs.python.org/3/library/importlib.html#importlib.abc.Loader.load_module)

**Pull request merged**